### DISCUSSION TITLE
amq.topic access is refused with the reason of "failed_connect"

### DISCUSSION THREAD

**Discussion by ssm6498** [rabbitmq/rabbitmq-server#11828] [Other] [answered]
we are getting below errors in between for a specific user, what can be the reson and what is the solution:

channel is already closed due to channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - write access to exchange 'amq.topic' in vhost '' refused for user '', backend rabbit_auth_backend_cache returned an error: {failed_connect, ..., class-id=60, method-id=40)

**Comment by michaelklishin** [comment]
@ssm6498 if you are urgently looking for help, consider providing detailed logs and not just a few lines

> ACCESS_REFUSED - write access to exchange 'amq.topic' in vhost '' refused for user ''

> returned an error: {failed_connect

You are using `rabbitmq_auth_backend_cache` with a different plugin, likely LDAP or HTTP, which returned "failed_connect". So something somewhere in the authN/authZ backend chain fails to connect somewhere. We cannot possibly know what it is or what may be the solution since you have left out the details and we do not have access to your infrastructure.

**Comment by michaelklishin** [comment, marked as answer]
A quick search for `failed_connect` suggests the [underlying backend is OAuth 2](https://github.com/search?q=repo%3Arabbitmq%2Frabbitmq-server%20failed_connect&type=code).

OAuth 2 was not meant to be used in combination with the caching backend since it generally won't issue a new request once it has a token that's not near its expiration time.

Per our community support policy [we will not troubleshoot OAuth 2 for non-paying users](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#exceptions-question-that-will-be-ignored). However, the is a dedicated [OAuth 2 troubleshooting guide](https://www.rabbitmq.com/docs/troubleshooting-oauth2), a dedicated section on [token expiration](https://www.rabbitmq.com/docs/oauth2#token-expiration) and [multiple detailed examples](https://www.rabbitmq.com/docs/oauth2-examples) for five or so identity providers
